compiler/src/function/main_function.rs

```diff
@@ -61,18 +61,34 @@ impl<'a, F: PrimeField, G: GroupType<F>> ConstrainedProgram<'a, F, G> {
             {
                 let input_variable = input_variable.get().borrow();
                 let name = input_variable.name.name.clone();
-                let input_option = input.get(&name).ok_or_else(|| {
-                    FunctionError::input_not_found(name.clone(), &function.span.clone().unwrap_or_default())
-                })?;
-                let input_value = self.allocate_main_function_input(
-                    cs,
-                    &input_variable.type_.clone(),
-                    &name,
-                    input_option,
-                    &function.span.clone().unwrap_or_default(),
-                )?;
-
-                // Store a new variable for every allocated main function input
+
+                let input_value = match (input.get(&name), input.get_constant(&name)) {
```

Comment:
Your code handles the following case incorrectly:
```rust
[main]
x: u8 = 1;

[constant]
x: u8 = 2;
```


Your code will execute the `(Some(input_option), _) => ...` branch for both values. 
The expected behavior should be to throw a "Duplicate input name" error of some sort. You can catch this case with a `(Some(input_option), Some(input_option)) => ...` branch at the beginning of the match.

I don't think we directly addressed this case in the proposal and it is good that we caught it now.

Reply:
Covered case with error.